src/tools/remote-test-client/tests/lib.rs

```diff
@@ -8,3 +8,18 @@ fn test_help() {
     let stdout = String::from_utf8(output.stdout.clone()).unwrap();
     assert!(stdout.trim().starts_with("Usage:"));
 }
+
+#[test]
+fn test_timeout() {
+    let mut cmd = assert_cmd::cargo::cargo_bin_cmd!();
+    cmd.env("TEST_DEVICE_CONNECT_TIMEOUT_SECONDS", "1");
+    cmd.env("TEST_DEVICE_ADDR", "0.0.0.0:6969");
+    cmd.args(["spawn-emulator", "riscv64-unknown-linux-gnu", "./"]);
+    cmd.arg(std::env::temp_dir());
+
+    let assert = cmd.assert().failure();
+    let output = assert.get_output();
+
+    let stderr = String::from_utf8(output.stderr.clone()).unwrap();
+    assert!(stderr.contains("Gave up trying to connect to test device"));
```

Comment:
when this fails it will be very helpful to see all of stderr, so let's


```suggestion
    let pass_msg = "Gave up trying to connect to test device";
    assert!(stderr.contains(pass_msg), "Could not find `{pass_msg}` in `{stderr}`");
```